src/server/system_services/master_key_manager.js

```diff
@@ -43,38 +48,68 @@ class MasterKeysManager {
 
 
     get_root_key_id() {
-        return ROOT_KEY;
+        const { NOOBAA_ROOT_SECRET } = process.env;
+        return NOOBAA_ROOT_SECRET ? ROOT_KEY : undefined;
     }
 
     is_root_key(root_key_id) {
         return root_key_id && root_key_id.toString() === ROOT_KEY.toString();
     }
 
+    get_current_root_key_id() {
+        return this.active_root_key && this.active_root_key.toString();
+    }
+
     /**
      * 
      * @returns {Object} 
      */
     get_root_key() {
-        return this.resolved_master_keys_by_id[ROOT_KEY];
+        return this.resolved_master_keys_by_id[this.get_root_key_id() || this.get_current_root_key_id()];
```

Comment:
this.get_root_key_id() and this.get_current_root_key_id() should be united to the same function